lib/item.php

```diff
@@ -0,0 +1,202 @@
+<?php
+/**
+ * Copyright (c) 2015 Victor Dubiniuk <victor.dubiniuk@gmail.com>
+ * This file is licensed under the Affero General Public License version 3 or
+ * later.
+ * See the COPYING-README file.
+ */
+
+namespace OCA\Files_Antivirus;
+
+use OCA\Files_Antivirus\Status;
+
+class Item {
+	protected $id;
+	protected $view;
+	protected $path;
+	protected $status;
+	
+	protected $fileHandle;
+	protected $chunkSize;
+	
+	protected $isValidSize;
+	
+	public function __construct($view, $path, $id = null) {
+		if (!is_object($view)){
+			$this->logError('Can\'t init filesystem view.', $id, $path);
+			throw new \RuntimeException();
+		}
+		
+		if(!$view->file_exists($path)) {
+			$this->logError('File does not exist.', $id, $path);
+			throw new \RuntimeException();
+		}
+		
+		if (is_null($id)){
+			$this->id = $view->getFileInfo($path)->getId();
+		} else {
+			$this->id = $id;
+		}
+		
+		$this->view = $view;
+		$this->path = $path;
+		
+		$this->isValidSize = $view->filesize($path) > 0;
+		
+		$application = new \OCA\Files_Antivirus\AppInfo\Application();
+		$config = $application->getContainer()->query('Appconfig');
+		$this->chunkSize = $config->getAvChunkSize();
+	}
+	
+	/**
+	 * Is this file good for scanning? 
+	 * @return boolean
+	 */
+	public function isValid() {
+		$isValid = !$this->view->is_dir($this->path) && $this->isValidSize;
+		return $isValid;
+	}
+	
+	/**
+	 * Reads a file portion by portion until the very end
+	 * @return string|boolean
+	 */
+	public function fread() {
+		if (!$this->isValid()) {
+			return;
+		}
+		if (is_null($this->fileHandle)) {
+			$this->getFileHandle();
+		}
+		
+		if (!is_null($this->fileHandle) && !$this->feof()) {
+			$chunk = fread($this->fileHandle, $this->chunkSize);
+			return $chunk;
+		}
+		return false;
+	}
+	
+	/**
+	 * Action to take if this item is infected
+	 * @param Status $status
+	 * @param boolean $isBackground
+	 */
+	public function processInfected(Status $status, $isBackground) {
+		if ($isBackground) {
+			$application = new \OCA\Files_Antivirus\AppInfo\Application();
+			$appConfig = $application->getContainer()->query('Appconfig');
+			$infectedAction = $appConfig->getInfectedAction();
+			if ($infectedAction == 'delete') {
+				$this->logError('Infected file deleted. ' . $status->getDetails());
+				$this->view->unlink($this->path);
+			} else {
+				$this->logError('File is infected. '  . $status->getDetails());
+			}
+		} else {
+			$this->logError('Virus(es) found: ' . $status->getDetails());
+			//remove file
+			$this->view->unlink($this->path);
+			Notification::sendMail($this->path);
+			$message = \OCP\Util::getL10N('files_antivirus')
+						->t(
+								"Virus detected! Can't upload the file %s", 
+								array(basename($this->path))
+							)
+			;
+			\OCP\JSON::error(array("data" => array( "message" => $message)));
+			exit();
```

Comment:
exit has to go away - but this is anyhow to be replace with a streamwrapper as discussed